features/git-town/configuration/configuration_initial.feature

```diff
@@ -0,0 +1,73 @@
+Feature: Initial configuration
+
+  Background:
+    Given I haven't configured Git Town yet
+
+
+  Scenario Outline: Running Git Town commands while Git Town is unconfigured
+    When I run `<COMMAND>` while allowing errors
+    Then I see
+      """
+      Git Town hasn't been configured for this repository.
+      Please run 'git town config --setup'.
+      Would you like to do that now? y/n
+      """
+
+    Examples:
+    | COMMAND            |
+    | git extract        |
+    | git hack           |
+    | git kill           |
+    | git pr             |
+    | git prune-branches |
+    | git repo           |
+    | git ship           |
+    | git sync           |
+    | git sync-fork      |
+
+
+  Scenario: Enter non-existent main branch
+    When I run `git town config --setup` and enter "nonexistent"
+    Then I see
+      """
+      Please enter the name of the main dev branch (typically 'master' or 'development'):
+
+        Error
+        There is no branch named 'nonexistent'
+
+      """
+    And Git Town is still not configured for this repository
```

Comment:
I disagree that is an "extreme edge case". This is the first interaction users will have with Git Town. I'd rather have this be fully functional and intuitive and not make me have to rerun commands. I can implement this in a separate PR if @allewun doesn't want to add it to this.